src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -510,21 +518,19 @@ private IEnumerable<sqlite3_stmt> PrepareAndEnumerateStatements()
             while (!string.IsNullOrEmpty(tail));
         }
 
-        private IEnumerable<sqlite3_stmt> ResetAndEnumerateStatements()
+        private void DisposePreparedStatements()
         {
-            foreach (var stmt in _preparedStatements)
+            if (DataReader != null)
             {
-                raw.sqlite3_reset(stmt);
-
-                yield return stmt;
+                DataReader.DisposeStatements = true;
+                DataReader = null;
```

Comment:
This is an exceptional case that only happens when you do one of the following with an open SqliteDataReader
* Change the command's Connection
* Change the command's CommandText
* Dispose the command

All SqliteDataReader's should be explicitly disposed by the user per the ADO.NET contract.

If there is no active reader, disposing the command will dispose the statements.